Rover/sailboat.cpp

```diff
@@ -552,3 +585,111 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+bool Sailboat::in_irons()
+{
+   // only enabled for separate mainsail and options bit set
+   if (! get_out_of_in_irons_in_auto_enabled()){
+       clear_in_irons();
+       return false;
+   }
+   // helper lambda functions
+
+   // do low level set in irons work
+   auto set_in_irons = [this]()
+   {
+       // fix the desired heading while in irons
+       in_irons_heading_rad =
+          (currently_tacking)
+             ? tack_heading_rad
+             : radians(rover.g2.wp_nav.oa_wp_bearing_cd() * 0.01)
+       ;
+
+       // calculate which way to set the rudders to rotate the boat
+       // towards desired heading as it is being blown backwards
+       const int in_irons_rotate_sign =
+        (wrap_PI(in_irons_heading_rad - rover.ahrs.get_yaw()) >= 0.0f)
+           ? Sailboat::rotate_clockwise
+           : Sailboat::rotate_anticlockwise
+       ;
+
+       // we want the boat to rotate so relax the sails
+       // to prevent them acting as a weather vane
+       relax_sails();
+       // set the rudder for in irons steering output
+       in_irons_rudder = in_irons_rotate_sign * Sailboat::in_irons_rudder_percent;
+       is_in_irons = true;
+       gcs().send_text(MAV_SEVERITY_INFO, "Sailboat: In Irons rotate onto %s",
+          (in_irons_rotate_sign == Sailboat::rotate_clockwise)? "port":"stbd");
+   };
+
+   auto going_backwards = []()-> bool{
+      float speed;
+      // return false if we don't have a valid speed (e.g no gps fix)
+      if (!rover.g2.attitude_control.get_forward_speed(speed)) {
+         return false;
+      }
+      return speed < -0.01f;
+   };
```

Comment:
The could just be a bool variable, no need to be a function since its only used once.